File: app/src/models/permission.ts
export interface Permission {
    id: number;
    employeeForename: string;
    employeeSurname: string;
    permissionType: PermissionType;
    grantedOn: string
}

export interface PermissionType{
    id: number;
    description: string;
}

export interface PermissionRequest {
    employeeForename: string;
    employeeSurname: string;
    permissionType: number;
}

export interface UpdatePermissionRequest extends PermissionRequest {
    id: number;
}

export interface PermissionTypeRequest {
    description: string;
}

File: api/Permissions.Domain/Models/Permission.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Permissions.Domain.Models;

public class Permission : Entity
{
    [StringLength(100)]
    public string EmployeeForename { get; set; }
    
    [StringLength(100)]
    public string EmployeeSurname { get; set; }
    
    [ForeignKey("PermissionTypeId")]
    public int PermissionTypeId { get; set; }
    
    public virtual PermissionType PermissionType { get; set; } = null!;
    
    public DateTime GrantedOn { get; set; }
}